src/buffered_input.jl

```diff
@@ -15,29 +15,26 @@ mutable struct BufferedInput
     end
 end
 
-
-# Read and buffer n more characters
-function __fill(bi::BufferedInput, bi_input::IO, n::Integer)
-    for _ in 1:n
-        c = eof(bi_input) ? '\0' : read(bi_input, Char)
-        i = bi.offset + bi.avail + 1
+# Read and buffer `n` more characters
+function buffer!(bi::BufferedInput, n::Integer)::Nothing
+    for i in bi.offset + bi.avail .+ (1:n)
```

Comment:
Does this amount to `(bi.offset + bi.avail) .+ (1:n)`? If so - can we add those parens to make it explicit? This mix of scalar addition + broadcast seems a little weird.
```

Reply:
Ok. It is not must but I make it explicit.